plugins/zabbix/alerta_zabbix.py

```diff
@@ -68,10 +70,12 @@ def status_change(self, alert, status, text):
                     LOG.debug('Zabbix: ack all failed, ack only the one event')
                     r = self.zapi.event.acknowledge(eventids=event_id, message='%s: %s' % (status, text), action=NO_ACTION)
                 except ZabbixAPIException as e:
+                    self.zapi.do_request('user.logout')
                     raise RuntimeError("Zabbix: ERROR - %s", e)
```

Comment:
Instead of logging out in line 73 and 78 just add a single `finally` to this `try-except` block that includes the logout. For example ...

```
            try:
                LOG.debug('Zabbix: ack all events for trigger...')
                r = self.zapi.event.acknowledge(eventids=event_ids, message='%s: %s' % (status, text), action=NO_ACTION)
            except ZabbixAPIException:
                try:
                    LOG.debug('Zabbix: ack all failed, ack only the one event')
                    r = self.zapi.event.acknowledge(eventids=event_id, message='%s: %s' % (status, text), action=NO_ACTION)
                except ZabbixAPIException as e:
                    raise RuntimeError("Zabbix: ERROR - %s", e)
            finally:
                self.zapi.do_request('user.logout')
```

The same applies to the changes below.